src/py21cmfast/src/ps.c

```diff
@@ -3984,6 +3984,14 @@ float adjust_redshifts_for_photoncons(
             adjusted_redshift = *redshift;
         }
         else {
+            // Initialise the photon non-conservation correction curve
+            // It is possible that for certain parameter choices that we can get here without initialisation happening.
+            // Thus check and initialise if not already done so
+            if(!photon_cons_allocated) {
+                determine_deltaz_for_photoncons();
+                photon_cons_allocated = true;
+            }
+
```

Comment:
Note, this is repetitive code with the other initialisation check further down. I thought about adding this earlier to only have this code once, but I'm not sure of the consequences of attempting to run this initialisation in other parts of this function (i.e. under certain conditions).